src/EventQueue.hs

```diff
@@ -36,28 +38,34 @@ emitDone chan = atomically $ writeTChan chan Done
 
 readEvents :: EventQueue -> IO [Event]
 readEvents chan = do
-  x <- atomically $ readTChan chan
-  threadDelay 100000
-  xs <- atomically $ emptyQueue
-  return (x : xs)
+  e <- atomically $ readTChan chan
+  unless (isKeyboardInput e) $ do
```

Comment:
I *do* prefer `when` over `unless`, but this was more about the fact that I would strongly prefer `isModifiedEvent` over `isKeyboardInput`, since it's just a discriminator for `Modified`.